package Protocol;
message Packet
{
	// Packettype
	enum PacketType { LoginRequest = 1; LoginResponse = 2; UserInformations = 3; }
	
	// types
	required PacketType packetType = 1;
	
	// packet types: Login
	optional LoginRequest requestLogin = 2;
	optional LoginResponse responseLogin = 3;
	
	// packet types: User informations
	optional UserInformations userinformations = 4;
}

// Sub Messages
message LoginRequest
{
	required string username = 1;
	required string password = 2;
}

message LoginResponse
{
	// Messagetype
	enum Type { Success = 1; LoginIncorect = 2; }

	// repsonse type
	required Type type = 1;
}

message UserInformations
{
	repeated Contact contact = 1;
}

message Contact
{
	// Packettype
	enum State { Offline = 0; Online = 1; }
	
	required int32 id = 1;
	required string username = 2;
	required State state = 3;
	required string group = 4;
}